cmd/prometheus/main.go

```diff
@@ -195,9 +193,6 @@ func (c *flagConfig) setFeatureListOptions(logger log.Logger) error {
 			case "metadata-wal-records":
 				c.scrape.AppendMetadata = true
 				level.Info(logger).Log("msg", "Experimental metadata records in WAL enabled, required for remote write 2.0")
-			case "new-service-discovery-manager":
```

Comment:
Should I add something like https://github.com/prometheus/prometheus/blob/d63f5b35df0f96b02961eff59205336812575689/cmd/prometheus/main.go#L252?

---

How about making the switch `fail` by default in case of an unknown flag?
It would require users to set them carefully but will help detecting misconfiguration/mistakes etc.

Reply:
why is that? to avoid being too demanding? because we could turn a blind eye on a misconfiguration... a warn log can easily be missed.

Reply:
Yep, it's a tradeoff, but I'd still want my prom to fail, but no one else is complaining about it, let's keep it like this.

Tell me if you think we should add a `This option for --enable-feature is now permanently enabled and therefore ...` for this.